src/sentry/scim/endpoints/teams.py

```diff
@@ -0,0 +1,213 @@
+                    target_user=member.user,
+                    event=AuditLogEntryEvent.MEMBER_JOIN_TEAM,
+                    data=omt.get_audit_log_data(),
+                )
+
+    def _remove_members_operation(self, request, operation, team):
+        try:
+            # grab the filter out of the brackets of the string that looks
+            # like so: members[userName eq "baz@sentry.io"]
+            parsed_filter = parse_filter_conditions(
+                re.search(r"\[(.*?)\]", operation["path"]).groups()[0]
+            )
+        except Exception:
+            # TODO: log parse error
+            raise ParseError(detail=SCIM_400_INVALID_FILTER)
+        member = OrganizationMember.objects.get(organization=team.organization, id=parsed_filter[0])
+
+        with transaction.atomic():
+            omt = OrganizationMemberTeam.objects.get(team=team, organizationmember=member)
+            self.create_audit_entry(
+                request=request,
+                organization=team.organization,
+                target_object=omt.id,
+                target_user=member.user,
+                event=AuditLogEntryEvent.MEMBER_LEAVE_TEAM,
+                data=omt.get_audit_log_data(),
+            )
+            omt.delete()
+
+    def _rename_team_opereation(self, request, operation, team):
+        serializer = TeamSerializer(
+            team,
+            data={
+                "slug": slugify(operation["value"]["displayName"]),
+            },
+            partial=True,
+        )
+        if serializer.is_valid():
+            team = serializer.save()
+            self.create_audit_entry(
+                request=request,
+                organization=team.organization,
+                target_object=team.id,
+                event=AuditLogEntryEvent.TEAM_EDIT,
+                data=team.get_audit_log_data(),
+            )
+
+    def patch(self, request, organization, team):
+        """
+        A SCIM Group PATCH request takes a series of operations to peform on a team.
+        It does them sequentially and if any of them fail no operations should go through.
+        The operations are add members, remove members, replace members, and rename team.
+        """
+        try:
+            with transaction.atomic():
+                for operation in request.data.get("Operations", []):
+                    op = operation["op"].lower()
+                    if op == GroupPatchOps.ADD and operation["path"] == "members":
+                        self._add_members_operation(request, operation, team)
+                    elif op == GroupPatchOps.REMOVE and "members" in operation["path"]:
+                        # the members op contains a filter string like so:
+                        # members[userName eq "baz@sentry.io"]
+                        self._remove_members_operation(request, operation, team)
+                    elif op == GroupPatchOps.REPLACE and not operation.get("path", None):
```

Comment:
This should be `RENAME`? I don't see any tests for rename

Reply:
so this is a bit confusing on the SCIM API side, both "rename group" and "replace members" are both done with the REPLACE op name, depending on the "path" (if path is "members", replace members, if no path, replace an attribute, of which the only one we support is displayName). I will rework the flow slightly to make this less confusing.

and i need to rework the integration tests a bit, have an assertion covering it, but doing too many things in test_group_flow.

Reply:
(i will refactor tests in a followup pr)